robots/cmd/kubevirt-job-copier/main_test.go

```diff
@@ -151,8 +161,231 @@ func Test_getSourceAndTargetRelease(t *testing.T) {
 	}
 }
 
+func TestCopyPeriodicJobsForNewProvider(t *testing.T) {
+	type args struct {
+		jobConfig                   *config.JobConfig
+		targetProviderReleaseSemver *querier.SemVer
+		sourceProviderReleaseSemver *querier.SemVer
+	}
+	tests := []struct {
+		name                                 string
+		args                                 args
+		wantUpdated                          bool
+		wantJobConfig                        *config.JobConfig
+		wantJobStatesToReportInSerialization bool
+	}{
+		{
+			name: "test reporterconfig not preserved with no job state to report",
```

Comment:
Not sure if the test case name reflects what we are checking here, before the changes in this PR a `reporter_config.slack.job_states_to_report: []` was not carried over and we ended up with `reporter_config.slack: {}`. This case covers that but the name says that the reporterconfig is not preserved, IMO we should make clear that the previous issue is solved and have a test that explicitly covers it, wdyt?

Reply:
Thanks for this catch! Added a description to remind us what we are testing and why. PTAL, thanks!